tests/scripts/components-basic-checks.sh

```diff
@@ -0,0 +1,162 @@
+# components-basic-checks.sh
```

Comment:
[INFORMATIVE] I compared the component assignments here and in #8226 as of bf47cf74374f8ae60d948ac1fac313a605afd3e9 and #9444 as of 8068f17c61b21400b654cde0daa796b6f4729b67.

```
comm -31 <(git grep -E -o '^component_\w+' upstream-public/pr/8226 -- 'tests/scripts/components*.sh' | sed 's!.*/\(.*\):\(.*\)!\2 \1!' | sort) <(git grep -E -o '^component_\w+' upstream-public/pr/9444 -- 'tests/scripts/components*.sh' | sed 's!.*/\(.*\):\(.*\)!\2 \1!' | sort)
```

No component present in both has moved to a different file.

The 3.6-only components are:

```
component_build_aes_via_padlock components-build-system.sh
component_build_full_psa_crypto_client_without_crypto_provider components-configuration-crypto.sh
component_build_psa_accel_alg_ecdh components-configuration-crypto.sh
component_build_psa_accel_alg_hkdf components-configuration-crypto.sh
component_build_psa_accel_alg_hmac components-configuration-crypto.sh
component_build_psa_accel_alg_md5 components-configuration-crypto.sh
component_build_psa_accel_alg_ripemd160 components-configuration-crypto.sh
component_build_psa_accel_alg_rsa_oaep components-configuration-crypto.sh
component_build_psa_accel_alg_rsa_pkcs1v15_crypt components-configuration-crypto.sh
component_build_psa_accel_alg_rsa_pkcs1v15_sign components-configuration-crypto.sh
component_build_psa_accel_alg_rsa_pss components-configuration-crypto.sh
component_build_psa_accel_alg_sha1 components-configuration-crypto.sh
component_build_psa_accel_alg_sha224 components-configuration-crypto.sh
component_build_psa_accel_alg_sha256 components-configuration-crypto.sh
component_build_psa_accel_alg_sha384 components-configuration-crypto.sh
component_build_psa_accel_alg_sha512 components-configuration-crypto.sh
component_build_psa_accel_key_type_rsa_key_pair components-configuration-crypto.sh
component_build_psa_accel_key_type_rsa_public_key components-configuration-crypto.sh
component_test_default_psa_crypto_client_without_crypto_provider components-configuration-crypto.sh
component_test_full_no_bignum components-configuration-crypto.sh
component_test_full_no_cipher_no_psa_crypto components-configuration-crypto.sh
component_test_full_no_cipher_with_psa_crypto components-configuration-crypto.sh
component_test_full_no_cipher_with_psa_crypto_config components-configuration-crypto.sh
component_test_psa_crypto_config_accel_hash_keep_builtins components-configuration-crypto.sh
component_test_psa_crypto_rsa_no_genprime components-configuration-crypto.sh
component_test_tls1_2_default_cbc_legacy_cbc_etm_cipher_only_use_psa components-configuration-tls.sh
component_test_tls1_2_default_cbc_legacy_cipher_only_use_psa components-configuration-tls.sh
component_test_tls1_2_default_stream_cipher_only_use_psa components-configuration-tls.sh
```

I agree with these assignments except `component_build_aes_via_padlock` as noted in another comment.